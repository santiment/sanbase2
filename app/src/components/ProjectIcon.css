.project-icon-default {
  width: 16px;
  display: inline-block;
}
