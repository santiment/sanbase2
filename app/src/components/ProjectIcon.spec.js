/* eslint-env jest */
import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { ProjectIcon } from './ProjectIcon'

describe('ProjectIcon component', () => {
  it('(smoke) it should render correctly', () => {
    const icon = shallow(<ProjectIcon name='Cofound.it' />)
    expect(toJson(icon)).toMatchSnapshot()
    expect(icon.prop('src')).toEqual('cofound-it.png')
  })

  it('SAN icon should render correctly', () => {
    const icon = shallow(<ProjectIcon name='Santiment' />)
    expect(toJson(icon)).toMatchSnapshot()
    expect(icon.prop('src')).toEqual('santiment.png')
  })

  it('DAO.Casino icon should render correctly', () => {
    const icon = shallow(<ProjectIcon name='DAO.Casino' />)
    expect(toJson(icon)).toMatchSnapshot()
    expect(icon.prop('src')).toEqual('dao-casino.png')
  })
})
