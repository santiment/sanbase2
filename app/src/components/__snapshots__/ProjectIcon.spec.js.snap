// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectIcon component (smoke) it should render correctly 1`] = `
<img
  alt="Cofound.it"
  height={16}
  src="cofound-it.png"
  width={16}
/>
`;

exports[`ProjectIcon component DAO.Casino icon should render correctly 1`] = `
<img
  alt="DAO.Casino"
  height={16}
  src="dao-casino.png"
  width={16}
/>
`;

exports[`ProjectIcon component DefaultIcon should render correctly 1`] = `
<ProjectIcon
  name="Any Not Available Name"
  size={16}
>
  <Component
    size={16}
  >
    <span
      className="project-icon-default"
      width={16}
    />
  </Component>
</ProjectIcon>
`;

exports[`ProjectIcon component SAN icon should render correctly 1`] = `
<img
  alt="Santiment"
  height={16}
  src="santiment.png"
  width={16}
/>
`;
